### PR TITLE
Introduce single-node Terraform example

### PR DESCRIPTION
**Pull request by lucendio**
* inspired by the multi-node example
* since the TF code already requires 0.12, I took the opportunity to refactor
  the template creation to unify two resources into one by utilizing the new
  recently introduced [`templatefile`](https://www.terraform.io/docs/configuration/functions/templatefile.html)

### CODE REVIEW

File: terraform/examples/multi-node/create-infrastructure.tf [hunk 1]
```diff
@@ -22,7 +22,9 @@ resource "hcloud_server" "node" {
   name        = "node${count.index}"
   image       = "ubuntu-18.04"
   server_type = "cx41"
-  ssh_keys    = ["hetznerssh-key"]
+  ssh_keys    = [
+    hcloud_ssh_key.default.name
```
Comment (jschaul) [inline terraform/examples/multi-node/create-infrastructure.tf]: There are a few warnings and errors here on terraform v0.12.17:

```
Warning: Interpolation-only expressions are deprecated

  on create-infrastructure.tf line 17, in resource "hcloud_ssh_key" "default":
  17:   public_key = "${file("secrets/myssh-key")}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.


Warning: Interpolation-only expressions are deprecated

  on create-infrastructure.tf line 112, in resource "null_resource" "vpnkube":
 112:   count = "${length(hcloud_server.node)}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.

(and 5 more similar warnings elsewhere)


Warning: Interpolation-only expressions are deprecated

  on create-infrastructure.tf line 17, in resource "hcloud_ssh_key" "default":
  17:   public_key = "${file("secrets/myssh-key")}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.


Warning: Interpolation-only expressions are deprecated

  on create-infrastructure.tf line 112, in resource "null_resource" "vpnkube":
 112:   count = "${length(hcloud_server.node)}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.

(and 5 more similar warnings elsewhere)


Error: Error in function call

  on create-infrastructure.tf line 17, in resource "hcloud_ssh_key" "default":
  17:   public_key = "${file("secrets/myssh-key")}"

Call to function "file" failed: no file exists at secrets/myssh-key.
```

File: terraform/examples/single-node/create-infrastructure.tf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  # Recommended: configure a backend to share terraform state
+  # See terraform documentation
+}
+
+# Configure the Hetzner Cloud Provider
+provider "hcloud" {
+  # You must have a HCLOUD_TOKEN environment variable set!
+}
+
+resource "hcloud_ssh_key" "default" {
```
Comment (jschaul) [inline terraform/examples/single-node/create-infrastructure.tf]: Same problems here:

```
terraform apply

Warning: Interpolation-only expressions are deprecated

  on create-infrastructure.tf line 17, in resource "hcloud_ssh_key" "default":
  17:   public_key = "${file("secrets/myssh-key")}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.


Warning: Interpolation-only expressions are deprecated

  on create-infrastructure.tf line 17, in resource "hcloud_ssh_key" "default":
  17:   public_key = "${file("secrets/myssh-key")}"

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.


Error: Error in function call

  on create-infrastructure.tf line 17, in resource "hcloud_ssh_key" "default":
  17:   public_key = "${file("secrets/myssh-key")}"

Call to function "file" failed: no file exists at secrets/myssh-key
```

**Comment by jschaul** [comment]
(that ssh key also doesn't exist, and there is no README/docs on how to generate it, please add that)

**Comment by jschaul** [comment]
I think this PR could use a rebase (and possibly a change/update regarding the recent changes to this repository)?

**Comment by lucendio** [comment]
TODO for myself:

1. Find out if this is still be useful (witht he new approach and all)
2. adapt the zero-copy-pasta approach

**Pull request closed without merging**